docs/csharp/tutorials/microservices.md

```diff
@@ -354,125 +358,130 @@ the machine parameters when you start, you only need to
 supply any changes. The changes here will be to include
 our application.
 
-In this first sample, we'll use the `onbuild` version of
-the RC2 image. This is the easiest way to create a working Docker
-environment. However, the image it creates is larger than necessary.
-This image include the dotnet core runtime, and the dotnet SDK. 
+In this first sample, we'll use the `latest` version of
+the dotnet image. This is the easiest way to create a working Docker
+environment. This image include the dotnet core runtime, and the dotnet SDK. 
+That makes it easier to get started and build, but does create a larger image.
 
-The next two lines load SQLite onto the machine:
-
-```
-RUN printf "deb http://ftp.us.debian.org/debian jessie main\n" >> /etc/apt/sources.list
-RUN apt-get -qq update && apt-get install -qqy sqlite3 libsqlite3-dev && rm -rf /var/lib/apt/lists/*
-```
-
-We're not using SQLite, but leave it in place for reference if you need it later.
-
-The next three lines setup your application:
+The next four lines setup and build your application:
 
 ```
 COPY . /app
 WORKDIR /app
 RUN ["dotnet", "restore"]
+RUN ["dotnet", "build"]
 ```
 
 This will copy the contents of the current directory to the docker VM, and restore
-all the packages.
+all the packages. Using the dotnet CLI means that the Docker image must include the
+.NET Core SDK. 
 
-The final lines of the file set the output port (80) and run the application:
+The final lines of the file set the tcp port (5000) this container 
+listens on and runs the application:
 
 ```
-EXPOSE 80
-ENTRYPOINT ["dotnet", "run"]
+EXPOSE 5000/tcp
+ENTRYPOINT ["dotnet", "run", "--server.urls", "http://0.0.0.0:5000"]
 ```
 
-Notice that this Dockerfile uses the dotnet cli to build and run your docker image.
-That's why the larger image is needed.
+The Dockerfile uses declares port 5000 here because it's the default port that an
+ASP.NET Core application will listen on. That same port is declared as the argument
+to the `--server.urls` command line option in the `dotnet run` command. Those must
+match for your container to listen on the proper port.
 
-Here are the steps to build the image and deploy it. The information below is 
-for the PowerShell CLI. Different shells will have slightly different syntax
-that will be highlighted below.
+## Building and running the image
 
-First, you have to create a new docker machine called 'weather-service':
+Let's build an image and run the service inside a Docker container. You build a container
+using the docker build command. Run the following command from the directory containing your code.
```

Comment:
<nit> Here it says "you build a container using the docker build command".  Later on the article says "When this command completes, you have a container ready to run...".  This wording is incorrect IMO and may cause some confusion.  When you build with Docker you technically produce an image.  When you run an image, you create a container.  If you want to think of it in terms of programming, an image is a class whereas a container is an instance of a class.  Here is an older article that explains the differences (first part of the article) - [http://paislee.io/how-to-automate-docker-deployments/](http://paislee.io/how-to-automate-docker-deployments/)